pkg/segment/writer/rawchecker.go

```diff
@@ -61,10 +65,12 @@ func ApplySearchToMatchFilterRawCsg(match *MatchFilter, col []byte) (bool, error
 	if match.MatchOperator == And {
 		var foundQword bool = true
 		if match.MatchType == MATCH_PHRASE {
-			regexp, err := match.GetRegexp()
-			if err != nil {
-				log.Errorf("ApplySearchToMatchFilterRawCsg: error getting match regex: %v", err)
-				return false, err
+			if !fast {
```

Comment:
It can try to fire the GetRegexp calculation for queries where regexp is nil/does not exist. Although GetRegexp will check for it. I thought this would be more clear.

Reply:
addressed in the latest commit.

Reply:
Good point